pre_commit/commands/run.py

```diff
@@ -37,7 +43,15 @@ def get_changed_files(new, old):
     )[1].splitlines()
 
 
-def get_filenames(args, include_expr, exclude_expr):
+def filter_filenames_by_types(filenames, types):
+    types = frozenset(types)
+    return tuple(
+        filename for filename in filenames
+        if tags_from_path(filename) >= types
```

Comment:
Do you think it makes sense for the list of tags to be an OR rather than an AND?

I'm wondering if there's a case where someone will do something like `types: [python, ruby]` and be surprised that it matches nothing rather than all python and ruby files.

I'm having trouble coming up with a scenario where I'd want to target files which have multiple tags... but I also can't really think of too many cases where I'd want to target both python and ruby but not just `text`, so maybe both cases are pretty rare?

My original (much more complicated) branch just checked for any intersection of the tags, but I didn't really spend much time considering the alternative.